### PR TITLE
mm_drv: tlb: Fix mapped page in bank calculation

### PR DESCRIPTION
**Pull request by jxstelter**
The initial implementation was broken during improvements. There was incorrect assumption that all pages are unmapped at initials state. In reality at the beginning whole memory is powered on, so we should mark all pages as mapped. Later in initialization code unused pages are unmapped and if after this some banks become empty (all pages unmapped), the power is switched off https://github.com/zephyrproject-rtos/zephyr/blob/290777b2b5dde44027ed5901b02f19223e5550aa/drivers/mm/mm_drv_intel_adsp_mtl_tlb.c#L679.

Signed-off-by: Jaroslaw Stelter <Jaroslaw.Stelter@intel.com>

### CODE REVIEW

**Review by marcinszkudlinski — DISMISSED** [review]

File: drivers/mm/mm_drv_bank.c [hunk 1]
```diff
@@ -21,18 +21,16 @@
...
 uint32_t sys_mm_drv_bank_page_mapped(struct mem_drv_bank *bank)
 {
 	bank->unmapped_pages--;
-	bank->mapped_pages++;
-	if (bank->mapped_pages > bank->max_mapped_pages) {
-		bank->max_mapped_pages = bank->mapped_pages;
-	}
+	if (bank->mapped_pages < bank->max_mapped_pages)
+		bank->mapped_pages++;
 	return bank->mapped_pages;
```
Comment (tmleman) [inline drivers/mm/mm_drv_bank.c:34]: Why we are always decrementing `unmapped_pages` value, but `mapped_pages` is incremented only when we didn't get to the max value?

Comment (jxstelter) [reply drivers/mm/mm_drv_bank.c]: The values are determined by HW definitions. Adding asserts.

**Review by tmleman — APPROVED** [review]

File: include/zephyr/drivers/mm/mm_drv_bank.h [hunk 2]
```diff
@@ -32,7 +34,9 @@ struct mem_drv_bank {
  *
  * The driver may wish to track various information about the memory banks
  * that it uses. This routine will initialize a generic structure containing
- * that information.
+ * that information. Since at the power on all memory banks are switched on
+ * it will start with all pages mapped. In next phase of driver initialization
+ * unused pages will be unmapped.
```
Comment (ceolin) [inline include/zephyr/drivers/mm/mm_drv_bank.h:39]: Which part of the driver initialization unused pages are unmapped ? I saw this being doing only in the test. This last sentence seems to be more related with the test or how you are using this driver than with the driver behavior itself.

**Review by ceolin — APPROVED** [review]
Changes look good, just a minor comment  about the documentation change.

**Review by dcpleung — APPROVED** [review]

**Pull request merged**